src/query/query_manager.c

```diff
@@ -2299,11 +2275,12 @@ qmgr_add_modified_class (THREAD_ENTRY * thread_p, const OID * class_oid_p)
 
   tran_index = LOG_FIND_THREAD_TRAN_INDEX (thread_p);
   tran_entry_p = &qmgr_Query_table.tran_entries_p[tran_index];
-
-  if (tran_entry_p->modified_classes_p == NULL
-      && (tran_entry_p->modified_classes_p = qmgr_allocate_oid_block (thread_p)) == NULL)
+  if (tran_entry_p->modified_classes_p == NULL)
     {
-      return;
+      tran_entry_p->modified_classes_p = (OID_BLOCK_LIST *) malloc (sizeof (OID_BLOCK_LIST));
+      assert (tran_entry_p->modified_classes_p != NULL);
```

Comment:
How about error handling as below?
```suggestion
      if (tran_entry_p->modified_classes_p == NULL)
    {
	  er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_OUT_OF_VIRTUAL_MEMORY, 1, sizeof (OID_BLOCK_LIST));
	  return;
    }
```

Reply:
It's not recoverable. The system is corrupted something.